### ISSUE TITLE
AdminCommandHandlers: better arg checking

### ISSUE THREAD

**Issue by MikeJeffers** [OpenPerpetuum/PerpetuumServer#229]
https://github.com/OpenPerpetuum/PerpetuumServer/blob/Development/src/Perpetuum/Services/Channels/ChatCommands/AdminCommandHandlers.cs#L118
This and the TODOs following (and basically any command that checks and TryParses some command) deserves:
A) to work properly, note that often `err` is not checked or is only checked on the latest set value (bug!)
B) a better way to configure req/opt arguments and tryParse/type check them. (Stretch)

**Comment by MikeJeffers** [comment]
Especially on commands that can get you stuck or bugged out.
Checking zoneid params for valid zoneid, posistions within island size domain, etc.

**Comment by Veritania** [comment]
Hello!
I've opened PR #281 to check if you think this approach could add value.

The idea is to split the required parameters validation from the command-specific logic.

So the existing `AdminCommandHandlers` class will assign and validate the given data parameters.
It will then send the validated values to the newly created `AdminCommands` class which would eventually contain all the admin commands as separate methods.

These separate methods (  `zoneAddDecor`  in this example ) will then handle the command-specific logic within them, including checking for optional parameters that might have been provided.
This is also where it would validate if the parameters are actually valid, ex: we did get an int for zoneId, but is it a valid/existing zoneId?

I haven't looked through all the commands, so I'm not sure if this is applicable to all of them.
I'm also unsure if this is even a good approach, visually it seems a bit bloated to declare a bunch of empty variables, then assign and validate them, and then pass all of them to a method.
